public/angular/js/controllers/reports/index.js

```diff
@@ -372,14 +375,19 @@ angular.module('Aggie')
       // Pick one of the sources that has a media type. For now, it happens that
       // if a report has multiple sources, they all have the same type, or are
       // deleted
-      for (var i = 0; i < report._sources.length; i++) {
-        var sourceId = report._sources[i];
-        var source = $scope.sourcesById[sourceId];
-        if (source && $scope.mediaOptions[source.media] !== -1) {
-          return source.media + '-source';
+
+      if (report.metadata.platform === "Facebook") {
```

Comment:
Why does this exception from the for loop have to exist for Facebook?

Reply:
The exception is to handle the fact Facebook is a 'sub-source' within Crowdtangle, and to show the Facebook logo accordingly instead of the CrowdTangle logo which is in the source.media listing.